python/ray/_private/runtime_env/nsight.py

```diff
@@ -0,0 +1,132 @@
+import os
+import logging
+import subprocess
+import copy
+from typing import Tuple, List, Dict, Optional
+
+from ray._private.runtime_env.context import RuntimeEnvContext
+from ray._private.runtime_env.plugin import RuntimeEnvPlugin
+from ray._private.utils import (
+    try_to_create_directory,
+)
+
+default_logger = logging.getLogger(__name__)
+
+NSIGHT_DEFAULT_CONFIG = {
+    "-t": "cuda,cudnn,cublas,nvtx",
+    "-o": "'worker_process_%p'",
+    "--cudabacktrace": "all",
+    "--stop-on-exit": "true",
+}
+
+
+def check_nsight_script(nsight_config: Dict[str, str]) -> Tuple[bool, str]:
+    """
+    Function to check if the provided nsight options are
+    valid nsys profile options and if nsys profile is installed
+
+    The function returns:
+    - bool: True if the nsight_config is valid option
+    - str: error message if the nsight_config is invalid
+    """
+
+    # use empty as nsight report test filename
+    nsight_config_copy = copy.deepcopy(nsight_config)
```

Comment:
Hmm this function seems overkill. So this means you run the empty command to make sure if options are valid? Wouldn't error message supposed to be included in the runtime env failure message anyway?

Reply:
Yes, the issue was if we launch worker process with invalid nsight options, the Raylet just stuck and keep trying to relaunch worker process instead of terminate

Reply:
> Raylet just stuck and keep trying to relaunch worker process instead of terminate

Hmm this actually seems like a bug. Maybe we cannot catch the error if a worker process fails to start by a command error. Can you create an issue for this?

Reply:
for now, the workaround looks okay.

Reply:
Theoretically the cpp worker pool should be able to catch this (maybe it could be follow-up for you if you have time in the future)